src/asus.h

```diff
@@ -0,0 +1,212 @@
+#pragma once
+
+#include <linux/input.h>
+#include <stdint.h>
+
+#include "libratbag-private.h"
+#include "libratbag-hidraw.h"
+
+#define ASUS_QUIRK_DOUBLE_DPI 1 << 0
+#define ASUS_QUIRK_STRIX_PROFILE 1 << 1
+
+#define ASUS_PACKET_SIZE 64
+#define ASUS_BUTTON_ACTION_TYPE_KEY 0  /* keyboard key */
+#define ASUS_BUTTON_ACTION_TYPE_BUTTON 1  /* mouse button */
+#define ASUS_BUTTON_CODE_DISABLED 0xff  /* disabled mouse button */
+#define ASUS_STATUS_ERROR 0xffaa  /* invalid state/request, disconnected or sleeping */
+
+/* maximum number of buttons across all ASUS devices */
+#define ASUS_MAX_NUM_BUTTON 16
+
+/* maximum number of DPI presets across all ASUS devices */
+/* for 4 DPI devices: 0 - red, 1 - purple, 2 - blue (default), 3 - green */
+/* for 2 DPI devices: 0 - main (default), 1 - alternative */
+#define ASUS_MAX_NUM_DPI 4
+
+/* maximum number of LEDs across all ASUS devices */
+#define ASUS_MAX_NUM_LED 3
+
+/* base request */
+
+struct _asus_request {
+	uint16_t cmd;
+	uint8_t params[ASUS_PACKET_SIZE - 2];
+} __attribute__((packed));
+
+union asus_request {
+	struct _asus_request data;
+	uint8_t raw[ASUS_PACKET_SIZE];
+};
+
+/* base response */
+
+struct _asus_response {
+	uint16_t code;
+	uint8_t results[ASUS_PACKET_SIZE - 2];
+} __attribute__((packed));
+
+union asus_response {
+	struct _asus_response data;
+	uint8_t raw[ASUS_PACKET_SIZE];
+};
+
+/* current profile ID and firmware info */
+
+struct asus_profile_data {
+	unsigned int profile_id;
```

Comment:
..https://github.com/libratbag/libratbag/pull/1250/files#r993419635

Especially here since `unsigned int` may be a different thing depending on target platform.